pkg/devfile/generator/utils.go

```diff
@@ -245,7 +251,115 @@ func getPodTemplateSpec(podTemplateSpecParams podTemplateSpecParams) *corev1.Pod
 		},
 	}
 
-	return podTemplateSpec
+	if needsPodOverrides(globalAttributes, components) {
+		patchedPodTemplateSpec, err := applyPodOverrides(globalAttributes, components, podTemplateSpec)
+		if err != nil {
+			return nil, err
+		}
+		patchedPodTemplateSpec.ObjectMeta = podTemplateSpecParams.ObjectMeta
+		podTemplateSpec = patchedPodTemplateSpec
+	}
+
+	return podTemplateSpec, nil
+}
+
+// needsPodOverrides returns true if PodOverridesAttribute is present at Devfile or Container level attributes
+func needsPodOverrides(globalAttributes attributes.Attributes, components []v1.Component) bool {
+	if globalAttributes.Exists(PodOverridesAttribute) {
+		return true
+	}
+	for _, component := range components {
+		if component.Attributes.Exists(PodOverridesAttribute) {
+			return true
```

Comment:
can you add a test case where we have only component attribute and no global attribute to cover this?